lib/board.rb

```diff
@@ -14,25 +15,20 @@ def width
     @grid.first.length
   end
 
-  def paint!(row, col)
-    @grid[row][col].paint!
-    agent_is_player
+  def paint!(row, col, player)
+    @grid[row][col].paint!(player)
+    @current_player = (player == :player) ? :cpu : :player
   end
 
-  def agent_is_cpu
-    @agent = :cpu
```

Comment:
We often use the word `is` to describe conditions. So if I see `agent_is_cpu` I think `agent_is_cpu?` and I expect to get a boolean. That check would look like equality in the implementation (`==`), but this is assignment (`=`); I might not even notice that and — especially since the expression has a truthy value — the results could be very confusing!

The most general style guideline is to make methods that are actions verbs. Something like `set_agent_to_cpu` or `set_agent(:cpu)`. Then when you read the verb you know it's _doing something_ rather than describing something.

Reply:
I feel like we talked about this once already, or I read it somewhere. So, methods that change state might have more active language, whereas methods that describe a condition might use more declarative language. That makes total sense.